### ISSUE TITLE
return value of "java.util.function.Function.apply(Object)" is null when attempting to install plugin(s)

### ISSUE THREAD

**Issue by jakeogh** [eclipse-pde/eclipse.pde#1535]
Eclipse Version: 2024-12 (4.34.0)
Build id: 20241128-0757

While attempting to install `PEx_for_Kinetis_3.0.0.zip` I get:
```
An error occurred while collecting items to be installed
session context was:(profile=epp.package.cpp, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Cannot invoke "java.lang.Comparable.compareTo(Object)" because the return value of "java.util.function.Function.apply(Object)" is null
Cannot invoke "java.lang.Comparable.compareTo(Object)" because the return value of "java.util.function.Function.apply(Object)" is null
```

This happens when I launch eclipse as root as well. The prequisite plugin `freescale_updater.zip` installs without issue. I suspect this problem is not related to the specific plugin, becuase there is a open issue on the old bugtracker with the same error attempting to install a different plugin: https://bugs.eclipse.org/bugs/show_bug.cgi?id=583494 (the 2nd comment on that bug is from me).

Attempting the same install on a slightly older Eclipse:
Eclipse Version: 2023-06 (4.28.0)
Build id: 20230608-1333

I get what looks like a related error:
```
An error occurred while collecting items to be installed
session context was:(profile=epp.package.parallel, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Cannot invoke "String.compareTo(String)" because the return value of "org.eclipse.equinox.internal.p2.metadata.InstallableUnit.getId()" is null
Cannot invoke "String.compareTo(String)" because the return value of "org.eclipse.equinox.internal.p2.metadata.InstallableUnit.getId()" is null
```

In both cases, I started with a empty workspace and no other non-default plugins installed.

**Comment by laeubi** [comment]
Do you see any exceptions (e.g. error log or eclipse log) together with this? Beside this I think it is more a P2 thing than PDE.

**Comment by jakeogh** [comment]
`PEx_for_Kinetis_3.0.0.zip` is contained within the zip file `PEx_for_Kinetis_3.0.0_Install_into_Eclipse_4.4_Unzip_me.zip` which can be obtained from:
https://www.nxp.com/design/design-center/software/development-software/processor-expert-software/processor-expert-software-microcontroller-driver-suite:PE_DRIVER_SUITE?amp;tid=PEH

under the download header `PEx-KINETIS-v3.0-PLUGIN-ECLIPSE_4.4`

I followed the instructions `PEx_for_Kinetis_Installation_Guide.pdf` which is also contained within `PEx_for_Kinetis_3.0.0_Install_into_Eclipse_4.4_Unzip_me.zip`

I dont see anything interesting on STDOUT, where do I find additional logs?
`~/.eclipse $ find . -iname "*.log"` returned nothing.

**Comment by jakeogh** [comment]
Going back to:
Version: 2022-12 (4.26.0)
Build id: 20221201-1913

The `cache/content-1179742082.jar at line 5213, column 45: Missing required attribute in "unit": id` error still appears on `PEx_for_Kinetis_3.0.0.zip` install, but the instillation succeeds.

**Comment by merks** [comment]
Yes, this looks like a p2 problem.  Though that being said, it looks more like a problem that the update site is corrupt because it's widely assumed that the ID is non-null.   Perhaps the deserializer should omit such things, or map null to "".  But then what else should it all do to turn invalid input into not-so-invalid data?  Better would be to fix the update site, or add id="" to that file at the point at which it complains if this is some content for which the folks have stopped providing updates.
